flocker/node/_deploy.py

```diff
@@ -196,90 +196,89 @@ def run(self, deployer):
 
 
 @implementer(IStateChange)
-@attributes(["volume"])
-class CreateVolume(object):
+@attributes(["dataset"])
+class CreateDataset(object):
     """
-    Create a new locally-owned volume.
+    Create a new locally-owned dataset.
 
-    :ivar AttachedVolume volume: Volume to create.
+    :ivar Dataset dataset: Dataset to create.
     """
     def run(self, deployer):
         volume = deployer.volume_service.get(
-            name=_to_volume_name(self.volume.dataset.dataset_id),
-            size=VolumeSize(maximum_size=self.volume.dataset.maximum_size)
+            name=_to_volume_name(self.dataset.dataset_id),
+            size=VolumeSize(maximum_size=self.dataset.maximum_size)
         )
         return deployer.volume_service.create(volume)
 
 
 @implementer(IStateChange)
-@attributes(["volume"])
-class ResizeVolume(object):
+@attributes(["dataset"])
+class ResizeDataset(object):
     """
-    Resize an existing locally-owned volume.
+    Resize an existing locally-owned dataset.
 
-    :ivar AttachedVolume volume: Volume to resize.
+    :ivar Dataset dataset: Dataset to resize.
     """
     def run(self, deployer):
-        dataset = self.volume.manifestation.dataset
         volume = deployer.volume_service.get(
-            name=_to_volume_name(dataset.dataset_id),
-            size=VolumeSize(maximum_size=dataset.maximum_size)
+            name=_to_volume_name(self.dataset.dataset_id),
+            size=VolumeSize(maximum_size=self.dataset.maximum_size)
         )
         return deployer.volume_service.set_maximum_size(volume)
 
 
 @implementer(IStateChange)
-@attributes(["volume"])
-class WaitForVolume(object):
+@attributes(["dataset"])
+class WaitForDataset(object):
```

Comment:
This seems like the right thing for now.  I wonder if we'll want to split this into `WaitForPrimaryManifestation` and `WaitForReplica` later, though.